sec_certs/dataset/cve.py

```diff
@@ -147,9 +147,35 @@ def from_json(cls, input_path: Union[str, Path]):
             dset = json.load(handle, cls=CustomJSONDecoder)
         return dset
 
-    def get_cve_ids_for_cpe_uri(self, cpe_uri: str) -> Optional[Set[str]]:
+    def _get_cve_ids_for_cpe_uri(self, cpe_uri: str) -> Optional[Set[str]]:
         return self.cpe_to_cve_ids_lookup.get(cpe_uri, None)
 
+    def _get_cves_from_exactly_matched_cpes(self, cpe_matches: set[str]) -> set[str]:
+        cve_ids: set[str] = set()
+
+        for cpe_match in cpe_matches:
+            if cve_set := self._get_cve_ids_for_cpe_uri(cpe_match):
+                cve_ids.update(cve_set)
+
+        return cve_ids
+
+    def _get_cves_from_cpe_configurations(self, cpe_matches: set[str]) -> set[str]:
+        cves: set[str] = set()
+        cves_with_vulnerable_configurations = [cve for cve in self if cve.vulnerable_cpe_configurations]
+
+        for cve in cves_with_vulnerable_configurations:
+            if any([cpe_configuration.match(cpe_matches) for cpe_configuration in cve.vulnerable_cpe_configurations]):
+                cves.add(cve.cve_id)
+
+        return cves
+
+    def get_cves_from_matched_cpes(self, cpe_matches: set[str]) -> set[str]:
```

Comment:
While this is premature, please comment the methods a bit before merging. This one is crucial to grasp, as it explains that we're matching using two different ways.